LearningRateControl.py

```diff
@@ -236,6 +236,8 @@ def calc_relative_error(self, old_epoch, new_epoch):
       relative_error = (new_error - old_error) / abs(old_error)
     else:
       relative_error = (new_error - old_error) / abs(new_error)
+    import math
+    assert not math.isnan(relative_error)
```

Comment:
This can happen in certain situations (some losses sometimes can result in inf, and that can be ok in some situations), and I think it should not error here.

Reply:
(Also, this is anyway not really related to your pull request.)

Reply:
Ok, I didn't know that this can happen in valid situations. Do you think a warning should be printed?

I added this here, because such a warning would have made me aware of the failed LR scheduling much sooner (as written in the issue, the loss couldn't be found, so the error became NaN, disabling the LR scheduling without any notice). But I can remove it/create a separate PR for this.

Reply:
Normally, the training should stop if there is a nan/inf. You should get the message "Model seems broken, got inf or nan final score", and then it stops (unless you set `stop_on_nonfinite_train_score`).
You explicitly have to set `stop_on_nonfinite_train_score = False` in the config to disable this. But if you do you, you actually also say that it inf/nan scores are valid.
Anyway, this is handled elsewhere, and should be handled elsewhere in any case, not by the LR scheduling. If you don't like any of the behavior, create a new issue and let's discuss over there, but remove it from here to keep us focused here.

Reply:
Interesting. I'll check why it didn't abort in my case. But I agree, I'll remove it here.